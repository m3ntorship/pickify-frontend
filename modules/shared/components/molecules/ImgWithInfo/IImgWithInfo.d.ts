import type { ReactElement } from 'react';

declare namespace IImgWithInfo {
  export interface IProps {
    classes?: string;
    children: ReactElement;
    isResponsive?: boolean;
  }

  export interface IImage {
    classes?: string;
    children?: ReactElement;
    isResponsive?: boolean;
    ImageSize:
      | 'extra-large'
      | 'extra-small'
      | 'large'
      | 'medium'
      | 'small'
      | 'super-large';
    variant?: 'anonymous' | 'filled' | 'notFilled';
    profilePic?: string;
    onAvatarClickHandler?: () => void;
    ImageVariant: 'avatar' | 'icon';
    isHidden?: boolean;
    imagePath?: string;
  }

  export interface ITitle extends IProps {
    classes?: string;
    children?: ReactElement | string;
    titleSize: 'large' | 'medium' | 'small';
    titlePath?: string;
  }

  export interface ISubTitle extends IProps {
    classes?: string;
    children: ReactElement | string;
    subTitleSize: 'medium' | 'small';
    subTitlePath?: string;
  }
}

export { IImgWithInfo };
