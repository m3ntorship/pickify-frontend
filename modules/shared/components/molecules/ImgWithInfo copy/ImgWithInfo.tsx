import React from 'react';
import type { ReactElement } from 'react';
import classNames from 'classnames';
import styles from './ImgWithInfoDemo.module.css';
import type { IImgWithInfoDemo } from './IImgWithInfoDemo';
import Avatar from '../../atoms/Avatar/Avatar';
// this line case an error when we ran the test, the error : cannot find module or it is corresponding type
// import { handleImgWithInfoVariant } from '@modules/shared/logic/imgWithInfoVariant/imgWithInfoVariant';
import { handleImgWithInfoVariant } from '../../../logic/imgWithInfoVariant/imgWithInfoVariant';

const ImgWithInfoDemo = ({
  classes,
  children,
}: IImgWithInfoDemo.IProps): ReactElement => {
  const boxClasses = classNames(styles.box, classes);
  return <div className={boxClasses}>{children}</div>;
};

const Image = ({
  classes,
  avatarSize = 'medium',
  ImageVariant,
  profilePic,
  onAvatarClickHandler,
  isHidden = false,
  children,
}: IImgWithInfoDemo.IProps): ReactElement => {
  const ImageClasses = classNames(styles['header-wrapper'], classes);
  return (
    <div className={ImageClasses}>
      <div className={styles.header}>
        {ImageVariant === 'avatar' && (
          <Avatar
            size={avatarSize}
            variant={handleImgWithInfoVariant(isHidden, profilePic)}
            profilePic={profilePic}
            onClick={onAvatarClickHandler}
          />
        )}
      </div>
      {ImageVariant === 'icon' && children}
    </div>
  );
};

const Info = ({ classes, children }: IImgWithInfoDemo.IProps): ReactElement => {
  const bodyClasses = classNames(classes);
  return <div className={bodyClasses}>{children}</div>;
};

ImgWithInfoDemo.Image = Image;
ImgWithInfoDemo.Info = Info;

export default ImgWithInfoDemo;
