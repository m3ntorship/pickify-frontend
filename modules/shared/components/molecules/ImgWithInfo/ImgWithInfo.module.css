.info-box {
  @apply flex;
}

.responsive-info-box {
  @apply flex-col md:flex-row items-center md:items-start;
}

.responsive-image {
  @apply mr-0 md:mr-4 mb-2 md:mb-0;
}

.info-wrapper {
  @apply flex flex-col justify-around;
}

.responsive-info-wrapper {
  @apply items-center sm:items-start;
}

.title {
  @apply font-normal;
}

.subTitle {
  @apply font-light	text-dark-grey;
}
