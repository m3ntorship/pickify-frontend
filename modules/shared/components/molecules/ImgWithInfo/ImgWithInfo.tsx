import React from 'react';
import type { ReactElement } from 'react';
import classNames from 'classnames';
import styles from './ImgWithInfo.module.css';
import type { IImgWithInfo } from './IImgWithInfo';
import Avatar from '../../atoms/Avatar/Avatar';
import { handleImgWithInfoVariant } from '../../../logic/imgWithInfoVariant/imgWithInfoVariant';

const ImgWithInfo = ({
  classes,
  children,
  isResponsive,
}: IImgWithInfo.IProps): ReactElement => {
  const infoClasses = classNames(
    styles['info-box'],
    {
      [styles['responsive-info-box']]: isResponsive,
    },
    classes,
  );
  return <div className={infoClasses}>{children}</div>;
};

const Image = ({
  classes,
  ImageSize,
  ImageVariant,
  profilePic,
  onAvatarClickHandler,
  isHidden = false,
  children,
  isResponsive,
}: IImgWithInfo.IImage): ReactElement => {
  const ImageClasses = classNames(
    {
      'w-10 h-10': ImageSize === 'medium',
      'w-5xl h-5xl': ImageSize === 'super-large',
    },
    {
      [styles['responsive-image']]: isResponsive,
      'mr-4': !isResponsive,
    },
    classes,
  );

  return (
    <div className={ImageClasses}>
      {ImageVariant === 'avatar' && (
        <Avatar
          size={ImageSize}
          variant={handleImgWithInfoVariant(isHidden, profilePic)}
          profilePic={profilePic}
          onClick={onAvatarClickHandler}
        />
      )}
      {ImageVariant === 'icon' && children}
    </div>
  );
};

const Info = ({
  classes,
  children,
  isResponsive,
}: IImgWithInfo.IProps): ReactElement => {
  const infoWrapperClasses = classNames(
    styles['info-wrapper'],
    {
      [styles['responsive-info-wrapper']]: isResponsive,
    },
    classes,
  );
  return <div className={infoWrapperClasses}>{children}</div>;
};

const Title = ({
  classes,
  children,
  titleSize,
}: IImgWithInfo.ITitle): ReactElement => {
  const titleClasses = classNames(
    styles.title,
    {
      'text-sm': titleSize === 'small',
      'text-base': titleSize === 'medium',
      'text-lg': titleSize === 'large',
    },
    classes,
  );
  return <h3 className={titleClasses}>{children}</h3>;
};

const SubTitle = ({
  classes,
  children,
  subTitleSize,
}: IImgWithInfo.ISubTitle): ReactElement => {
  const subTitleClasses = classNames(
    styles.subTitle,
    {
      'text-xs': subTitleSize === 'small',
      'text-sm': subTitleSize === 'medium',
    },
    classes,
  );
  return <h5 className={subTitleClasses}>{children}</h5>;
};

const MoreInfo = ({ classes, children }: IImgWithInfo.IProps): ReactElement => {
  const moreInfoClasses = classNames('mt-2', classes);
  return <div className={moreInfoClasses}>{children}</div>;
};

ImgWithInfo.Image = Image;
ImgWithInfo.Info = Info;
Info.Title = Title;
Info.SubTitle = SubTitle;
Info.MoreInfo = MoreInfo;

export default ImgWithInfo;
