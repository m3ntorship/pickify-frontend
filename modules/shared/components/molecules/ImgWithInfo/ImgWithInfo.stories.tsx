import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import ImgWithInfo from './ImgWithInfo';
import type { IImgWithInfo } from './IImgWithInfo';

export default {
  title: 'molecules/ImgWithInfo',
  component: ImgWithInfo,
  argTypes: {
    profilePic: {
      control: {
        type: 'text',
      },
    },
    title: {
      control: {
        type: 'text',
      },
    },
    subTitle: {
      control: {
        type: 'text',
      },
    },
  },
};

const Template: Story<IImgWithInfo.IProps> = (args): ReactElement => (
  <ImgWithInfo {...args} />
);
export const ImgWithInfoStory = Template.bind({});
ImgWithInfoStory.args = {
  isHidden: true,
  title: 'Ahmed Ayoub',
  subTitle: '3 months ago',
  description: '13/08/2021 06:09:11',
  variant: 'avatar',
};
