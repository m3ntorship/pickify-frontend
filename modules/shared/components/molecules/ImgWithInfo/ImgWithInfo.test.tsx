import React from 'react';
import renderer from 'react-test-renderer';
import ImgWithInfo from './ImgWithInfo';
import TextPoll from '../../icons/textPoll.svg';

describe('ImgWithInfo component', () => {
  it('should render ImgWithInfo parent component with text Hello', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <>Hello</>
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render responsive ImgWithInfo parent component with text Hello when we pass isResponsive={true}', () => {
    const tree = renderer
      .create(
        <ImgWithInfo isResponsive>
          <>Hello</>
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Image component', () => {
  it('should render Image with Avatar component with extra small profile picture', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Image
            ImageSize="extra-small"
            ImageVariant="avatar"
            profilePic="https://placeimg.com/640/480/any"
          />
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render Image with Avatar component with small profile picture', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Image
            ImageSize="small"
            ImageVariant="avatar"
            profilePic="https://placeimg.com/640/480/any"
          />
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render Image with Avatar component with medium profile picture', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Image
            ImageSize="medium"
            ImageVariant="avatar"
            profilePic="https://placeimg.com/640/480/any"
          />
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('should render Image with Avatar component with large profile picture', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Image
            ImageSize="large"
            ImageVariant="avatar"
            profilePic="https://placeimg.com/640/480/any"
          />
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render Image with Avatar component with extra large profile picture', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Image
            ImageSize="extra-large"
            ImageVariant="avatar"
            profilePic="https://placeimg.com/640/480/any"
          />
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render Image with Avatar component with super large profile picture', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Image
            ImageSize="super-large"
            ImageVariant="avatar"
            profilePic="https://placeimg.com/640/480/any"
          />
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render Image with Icon component with medium size', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Image ImageSize="medium" ImageVariant="icon">
            <TextPoll />
          </ImgWithInfo.Image>
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render Image with not filled avatar component when we do not pass profile picture', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Image ImageSize="medium" ImageVariant="avatar" />
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render Image with anonymous avatar component when we pass isHidden={true}', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Image
            ImageSize="medium"
            ImageVariant="avatar"
            isHidden
          />
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render Image with filled avatar component when we pass profile picture even if when we pass isHidden={true}', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Image
            ImageSize="medium"
            ImageVariant="avatar"
            isHidden
            profilePic="https://placeimg.com/640/480/any"
          />
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render responsive Image component when we pass isResponsive={true}', () => {
    const tree = renderer
      .create(
        <ImgWithInfo isResponsive>
          <ImgWithInfo.Image
            isResponsive
            ImageSize="medium"
            ImageVariant="avatar"
            profilePic="https://placeimg.com/640/480/any"
          />
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Info component', () => {
  it('should render ImgWithInfo parent with Info component contains text Hello', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Info>
            <>Hello</>
          </ImgWithInfo.Info>
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render ImgWithInfo parent with responsive Info component when we pass isResponsive={true}', () => {
    const tree = renderer
      .create(
        <ImgWithInfo isResponsive>
          <ImgWithInfo.Info isResponsive>
            <>Hello</>
          </ImgWithInfo.Info>
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Title component', () => {
  it('should render ImgWithInfo with Title component with small text', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Info.Title titleSize="small">
            Hello
          </ImgWithInfo.Info.Title>
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render ImgWithInfo with Title component with large text', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Info.Title titleSize="large">
            Hello
          </ImgWithInfo.Info.Title>
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render ImgWithInfo with Title component Anonymous text', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Info.Title titleSize="small" />
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('SubTitle component', () => {
  it('should render ImgWithInfo with SubTitle component with small text', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Info.SubTitle subTitleSize="small">
            Hello
          </ImgWithInfo.Info.SubTitle>
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render ImgWithInfo with SubTitle component with medium text', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Info.SubTitle subTitleSize="medium">
            Hello
          </ImgWithInfo.Info.SubTitle>
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('MoreInfo component', () => {
  it('should render ImgWithInfo with MoreInfo component with Hello text', () => {
    const tree = renderer
      .create(
        <ImgWithInfo>
          <ImgWithInfo.Info.MoreInfo>
            <>Hello</>
          </ImgWithInfo.Info.MoreInfo>
        </ImgWithInfo>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
