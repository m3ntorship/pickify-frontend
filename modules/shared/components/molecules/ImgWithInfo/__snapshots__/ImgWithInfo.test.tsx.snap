// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component should render Image with Avatar component with extra large profile picture 1`] = `
<div
  className="info-box"
>
  <div
    className="mr-4"
  >
    <div
      className="wrapper extra-large"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "inline-block",
            "margin": 0,
            "maxWidth": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "maxWidth": "100%",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNjQiIGhlaWdodD0iNjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style={
              Object {
                "border": "none",
                "display": "block",
                "margin": 0,
                "maxWidth": "100%",
                "padding": 0,
              }
            }
          />
        </div>
        <img
          className="absolute w-full h-full"
          decoding="async"
          onClick={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Image component should render Image with Avatar component with extra small profile picture 1`] = `
<div
  className="info-box"
>
  <div
    className="mr-4"
  >
    <div
      className="wrapper extra-small"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "inline-block",
            "margin": 0,
            "maxWidth": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "maxWidth": "100%",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style={
              Object {
                "border": "none",
                "display": "block",
                "margin": 0,
                "maxWidth": "100%",
                "padding": 0,
              }
            }
          />
        </div>
        <img
          className="absolute w-full h-full"
          decoding="async"
          onClick={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Image component should render Image with Avatar component with large profile picture 1`] = `
<div
  className="info-box"
>
  <div
    className="mr-4"
  >
    <div
      className="wrapper large"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "inline-block",
            "margin": 0,
            "maxWidth": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "maxWidth": "100%",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTYiIGhlaWdodD0iNTYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style={
              Object {
                "border": "none",
                "display": "block",
                "margin": 0,
                "maxWidth": "100%",
                "padding": 0,
              }
            }
          />
        </div>
        <img
          className="absolute w-full h-full"
          decoding="async"
          onClick={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Image component should render Image with Avatar component with medium profile picture 1`] = `
<div
  className="info-box"
>
  <div
    className="w-10 h-10 mr-4"
  >
    <div
      className="wrapper medium"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "inline-block",
            "margin": 0,
            "maxWidth": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "maxWidth": "100%",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAiIGhlaWdodD0iNDAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style={
              Object {
                "border": "none",
                "display": "block",
                "margin": 0,
                "maxWidth": "100%",
                "padding": 0,
              }
            }
          />
        </div>
        <img
          className="absolute w-full h-full"
          decoding="async"
          onClick={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Image component should render Image with Avatar component with small profile picture 1`] = `
<div
  className="info-box"
>
  <div
    className="mr-4"
  >
    <div
      className="wrapper small"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "inline-block",
            "margin": 0,
            "maxWidth": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "maxWidth": "100%",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzIiIGhlaWdodD0iMzIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style={
              Object {
                "border": "none",
                "display": "block",
                "margin": 0,
                "maxWidth": "100%",
                "padding": 0,
              }
            }
          />
        </div>
        <img
          className="absolute w-full h-full"
          decoding="async"
          onClick={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Image component should render Image with Avatar component with super large profile picture 1`] = `
<div
  className="info-box"
>
  <div
    className="w-5xl h-5xl mr-4"
  >
    <div
      className="wrapper super-large"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "inline-block",
            "margin": 0,
            "maxWidth": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "maxWidth": "100%",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iODgiIGhlaWdodD0iODgiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style={
              Object {
                "border": "none",
                "display": "block",
                "margin": 0,
                "maxWidth": "100%",
                "padding": 0,
              }
            }
          />
        </div>
        <img
          className="absolute w-full h-full"
          decoding="async"
          onClick={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Image component should render Image with Icon component with medium size 1`] = `
<div
  className="info-box"
>
  <div
    className="w-10 h-10 mr-4"
  >
    <svg
      data-file-name="SvgTextPoll"
    />
  </div>
</div>
`;

exports[`Image component should render Image with anonymous avatar component when we pass isHidden={true} 1`] = `
<div
  className="info-box"
>
  <div
    className="w-10 h-10 mr-4"
  >
    <div
      className="wrapper medium"
    >
      <svg
        className="absolute w-full h-full"
        data-file-name="SvgAnonymous"
        height={40}
        onClick={[Function]}
        width={40}
      />
    </div>
  </div>
</div>
`;

exports[`Image component should render Image with filled avatar component when we pass profile picture even if when we pass isHidden={true} 1`] = `
<div
  className="info-box"
>
  <div
    className="w-10 h-10 mr-4"
  >
    <div
      className="wrapper medium"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "inline-block",
            "margin": 0,
            "maxWidth": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "maxWidth": "100%",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAiIGhlaWdodD0iNDAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style={
              Object {
                "border": "none",
                "display": "block",
                "margin": 0,
                "maxWidth": "100%",
                "padding": 0,
              }
            }
          />
        </div>
        <img
          className="absolute w-full h-full"
          decoding="async"
          onClick={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Image component should render Image with not filled avatar component when we do not pass profile picture 1`] = `
<div
  className="info-box"
>
  <div
    className="w-10 h-10 mr-4"
  >
    <div
      className="wrapper medium"
    >
      <svg
        className="absolute w-full h-full"
        data-file-name="SvgNotFilled"
        height={40}
        onClick={[Function]}
        width={40}
      />
    </div>
  </div>
</div>
`;

exports[`Image component should render responsive Image component when we pass isResponsive={true} 1`] = `
<div
  className="info-box responsive-info-box"
>
  <div
    className="w-10 h-10 responsive-image"
  >
    <div
      className="wrapper medium"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "inline-block",
            "margin": 0,
            "maxWidth": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "maxWidth": "100%",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDAiIGhlaWdodD0iNDAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style={
              Object {
                "border": "none",
                "display": "block",
                "margin": 0,
                "maxWidth": "100%",
                "padding": 0,
              }
            }
          />
        </div>
        <img
          className="absolute w-full h-full"
          decoding="async"
          onClick={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "visibility": "hidden",
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ImgWithInfo component should render ImgWithInfo parent component with text Hello 1`] = `
<div
  className="info-box"
>
  Hello
</div>
`;

exports[`ImgWithInfo component should render responsive ImgWithInfo parent component with text Hello when we pass isResponsive={true} 1`] = `
<div
  className="info-box responsive-info-box"
>
  Hello
</div>
`;

exports[`Info component should render ImgWithInfo parent with Info component contains text Hello 1`] = `
<div
  className="info-box"
>
  <div
    className="info-wrapper"
  >
    Hello
  </div>
</div>
`;

exports[`Info component should render ImgWithInfo parent with responsive Info component when we pass isResponsive={true} 1`] = `
<div
  className="info-box responsive-info-box"
>
  <div
    className="info-wrapper responsive-info-wrapper"
  >
    Hello
  </div>
</div>
`;

exports[`MoreInfo component should render ImgWithInfo with MoreInfo component with Hello text 1`] = `
<div
  className="info-box"
>
  <div
    className="mt-2"
  >
    Hello
  </div>
</div>
`;

exports[`SubTitle component should render ImgWithInfo with SubTitle component with medium text 1`] = `
<div
  className="info-box"
>
  <h5
    className="subTitle text-sm"
  >
    Hello
  </h5>
</div>
`;

exports[`SubTitle component should render ImgWithInfo with SubTitle component with small text 1`] = `
<div
  className="info-box"
>
  <h5
    className="subTitle text-xs"
  >
    Hello
  </h5>
</div>
`;

exports[`Title component should render ImgWithInfo with Title component Anonymous text 1`] = `
<div
  className="info-box"
>
  <h3
    className="title text-sm"
  >
    Anonymous
  </h3>
</div>
`;

exports[`Title component should render ImgWithInfo with Title component with large text 1`] = `
<div
  className="info-box"
>
  <h3
    className="title text-lg"
  >
    Hello
  </h3>
</div>
`;

exports[`Title component should render ImgWithInfo with Title component with small text 1`] = `
<div
  className="info-box"
>
  <h3
    className="title text-sm"
  >
    Hello
  </h3>
</div>
`;
